pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -553,6 +547,21 @@ private void createPersistentSubscriptions() {
         checkReplicatedSubscriptionControllerState();
     }
 
+    private CompletableFuture<Void> removeOrphanReplicationCursors() {
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        List<String> replicationClusters = topicPolicies.getReplicationClusters().get();
+        for (ManagedCursor cursor : ledger.getCursors()) {
+            if (cursor.getName().startsWith(replicatorPrefix)) {
+                String remoteCluster = PersistentReplicator.getRemoteCluster(cursor.getName());
+                if (!replicationClusters.contains(remoteCluster)) {
```

Comment:
Why just judge that it is not in the topic policies and then remove it? Should still judge whether it is not in namespace policies?

Reply:
> Why just judge that it is not in the topic policies and then remove it? Should still judge whether it is not in namespace policies?

It is a `HierarchyValue`, see the following code

```java
  private void updateValue() {
        VALUE_UPDATER.updateAndGet(this, (preValue) -> {
            if (topicValue != null) {
                return topicValue;
            } else if (namespaceValue != null) {
                return namespaceValue;
            } else {
                return brokerValue;
            }
        });
    }
```